### PR TITLE
teach strip_tab_ansi() to strip ANSI CSO sequences

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/string-util.c [hunk 1]
```diff
@@ -703,7 +705,17 @@ char *strip_tab_ansi(char **ibuf, size_t *_isz, size_t highlight[2]) {
         assert(ibuf);
         assert(*ibuf);
 
-        /* Strips ANSI color and replaces TABs by 8 spaces */
+        /* This does three things:
+         *
+         * 1. Replaces TABs by 8 spaces
+         * 2. Strips ANSI color sequences, i.e. CSI '[' … 'm' sequences
+         * 3. Strips ANSI operating system sequences, i.e. CSO '[' … BEL sequences
```
Comment (filbranden) [inline src/basic/string-util.c]: I think it's `CSO ']' ... BEL sequences`, at least from reading the code...

File: src/test/test-strip-tab-ansi.c [hunk 2]
```diff
@@ -36,5 +38,13 @@ int main(int argc, char *argv[]) {
         assert_se(streq(p, "\x1B[waldo"));
         free(p);
 
+        assert_se(terminal_urlify_path("/etc/fstab", "i am a link", &urlified) >= 0);
+        assert_se(p = strjoin("something ", urlified, " something-else"));
+        printf("<%s>\n", p);
+        assert_se(strip_tab_ansi(&p, NULL, NULL));
+        printf("<%s>\n", p);
+        assert_se(streq(p, "something i am a link something-else"));
+        p = mfree(p);
```
Comment (keszybz) [inline src/test/test-strip-tab-ansi.c]: I think it'd be great to also add a test with partial sequences here, just to test that branch of the code too.

File: src/basic/string-util.c (hunk 1 above)
Comment (poettering) [reply src/basic/string-util.c]: Actually, ']' is already part of the CSO sequence itself, hence the existing comment is broken in even more ways, as a strict reading would hence mean ESC ']' '[' … BEL... which is of course bogus...

**Comment by poettering** [comment]
Force pushed a new version now, with the comment fixed and with one more test that checks behaviour on a truncated sequence. PTAL

**Comment by keszybz** [comment]
`140/269 test-strip-tab-ansi                     FAIL     0.00 s`

**Comment by poettering** [comment]
Oops. Forgot to commit my final fixes before pushing. Redone now, PTAL

**Pull request merged**